q2cli/tests/test_usage.py

```diff
@@ -0,0 +1,58 @@
+import pytest
+
+from q2cli.core.usage import CLIUsage
+from qiime2.core.testing.util import get_dummy_plugin
+
+
+params = [
+    (
+        'concatenate_ints',
+        'concatenate_ints_simple',
+        (
+            "# This example demonstrates basic usage.",
+            "qiime dummy-plugin concatenate-ints \\",
+            "    --i-ints1 ints_a.qza \\",
+            "    --i-ints2 ints_b.qza \\",
+            "    --i-ints3 ints_c.qza \\",
+            "    --p-int1 4 \\",
+            "    --p-int2 2 \\",
+            "    --o-concatenated-ints ints_d.qza",
+        ),
+    ),
+    (
+        'concatenate_ints',
+        'concatenate_ints_complex',
+        (
+            "# This example demonstrates chained usage (pt 1).",
+            "qiime dummy-plugin concatenate-ints \\",
+            "    --i-ints1 ints_a.qza \\",
+            "    --i-ints2 ints_b.qza \\",
+            "    --i-ints3 ints_c.qza \\",
+            "    --p-int1 4 \\",
+            "    --p-int2 2 \\",
+            "    --o-concatenated-ints ints_d.qza",
+            # TODO: Test to make sure there is additional \n here?
+            "# This example demonstrates chained usage (pt 2).",
+            "qiime dummy-plugin concatenate-ints \\",
+            "    --i-ints1 ints_d.qza \\",
+            "    --i-ints2 ints_b.qza \\",
+            "    --i-ints3 ints_c.qza \\",
+            "    --p-int1 41 \\",
+            "    --p-int2 0 \\",
+            "    --o-concatenated-ints concatenated_ints.qza",
+        ),
+    ),
+]
+
+
+@pytest.fixture
+def dummy_plugin():
+    return get_dummy_plugin()
+
+
+@pytest.mark.parametrize("action, example, exp", params)
```

Comment:
I'm not familiar with this feature of pytest - will it re-rerun this test multiple times with each test fixture? As long as a new instance of `CLIUsage` is constructed each time, you should be fine.

Reply:
`test_render` will run once for each tuple in the list of tuples `params`.  Each tuple defines the arguments `action`, `example`, and `exp` that get passed to `test_render`.  and yes, each run will create a new instance of `CLIUsage`.  It will also provide a fresh `dummy_plugin`, but if that's unnecessary I can broaden the scope of the `dummy_plugin` pytest fixture so the same `dummy_plugin` is used throughout the module.

Reply:
There is one pytest fixture in this module so far. It just sets up/tears down `dummy_plugin`.  Of course, set up/tear down happens once for each test (unless I change the scope of the fixture).  As of now, `test_render` runes twice, once for each of the two tuples in `params`.  The plan is to increase test cases by expanding `params` to cover all the test cases in `qiime2/tests/test_artifact_api.py`